contracts-v0.2/src/Aggregator.sol

```diff
@@ -128,20 +110,17 @@ contract Aggregator is Ownable, IAggregator, ITypeAndVersion {
 
     /**
      * @notice called by oracles when they have witnessed a need to update
-     * @param _roundId is the ID of the round this submission pertains to
      * @param _submission is the updated data that the oracle is submitting
      */
-    function submit(uint256 _roundId, int256 _submission) external {
-        bytes memory error = validateOracleRound(msg.sender, uint32(_roundId));
+    function submit(int256 _submission) external {
+        uint32 roundId = reportingRoundId + 1;
+        bytes memory error = validateOracleRound(msg.sender, uint32(roundId));
         require(error.length == 0, string(error));
 
-        oracleInitializeNewRound(uint32(_roundId));
-        recordSubmission(_submission, uint32(_roundId));
-        (bool updated, int256 newAnswer) = updateRoundAnswer(uint32(_roundId));
-        deleteRoundDetails(uint32(_roundId));
-        if (updated) {
-            validateAnswer(uint32(_roundId), newAnswer);
-        }
+        oracleInitializeNewRound(uint32(roundId));
+        recordSubmission(_submission, uint32(roundId));
+        updateRoundAnswer(uint32(roundId));
```

Comment:
The return values from `updateRoundAnswer` are not used anymore. Shouldn't we update the function as well?

Reply:
hmm! I see. Can we fully remove this function, then update `latestRoundID` inside of `Submit` function?

Reply:
Not now, you would have to make many other updates as well.